### PR TITLE
Add documentation for upgrading Angular dependencies.

### PR DESCRIPTION
**Pull request by bmd3k**

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -302,3 +302,53 @@ Sample upgrade: https://github.com/tensorflow/tensorboard/pull/5977
...
+    version that is compatible with the version of Angular.
+* `ngx-color-picker`
+  * Similarly, the latest version might be fine but you also might have to guess
+    at the most recent version that is compatible with the version fo Angular.
+
+Sample upgrades:
+  * Angular 13: https://github.com/tensorflow/tensorboard/pull/6063.
+  * Angular 14: https://github.com/tensorflow/tensorboard/pull/6066.
+
+The upgrade can be performed using the steps described
```
Comment (JamesHollyer) [inline DEVELOPMENT.md:350]: I canot find a #adding-updating-or-removing-frontend-dependencies section. I am assuming it exists, this works, and is useful.

**Review by JamesHollyer — APPROVED** [review]

File: DEVELOPMENT.md (hunk 1 above)
Comment (bmd3k) [reply DEVELOPMENT.md]: The anchor links are generated by GitHub based on the section titles.

Click on the "in this section" link here:

https://github.com/bmd3k/tensorboard/blob/angular-upgrade-doc/DEVELOPMENT.md

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -302,3 +302,53 @@ Sample upgrade: https://github.com/tensorflow/tensorboard/pull/5977
...
+  * For example, [lookup the available versions of `@angular/core`](https://www.npmjs.com/package/@angular/core)
+
+When upgrading Angular we generally must consider upgrading the following
+dependencies listed in `package.json`:
+
+* All `@angular/*` and `@angular-devkit/*` dependencies.
+  * Except `@angular/build-tooling`, for which we currently don't have any
+    upgrade policy/guidance. It is acceptable to leave this alone unless you
+    discover a need to upgrade it.
+  * Most of these should be upgraded to the ~same version. The easiest is to
```
[the remark this answers is not in the document]
Comment (bmd3k) [reply DEVELOPMENT.md:329]: Yes, agreed. I added this to the preamble before this list and moved the "in this section" link there, too.

File: DEVELOPMENT.md (hunk 1 above)
Comment (JamesHollyer) [reply DEVELOPMENT.md]: Makes sense. Thanks

**Pull request merged**